### PR TITLE
Be consistent about OnModelCreating as the terminology for fluent API

### LINKED ISSUE #22310: Update DisplayName for shared-type entity types to include the entity type name

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Use "navigation" rather than "navigation property"
Check usage of {foreignKey} and for correct quoting and use foreignKeyProperties where appropriate
Fix AnnotationNotFound, BadFilterDerivedType, CircularDependency
Update DisplayName for shared-type entity types to include the entity type name

Part of #7201
Fixes #22310

### CODE REVIEW

**Comment by AndriySvyryd** [comment]
@Pilchie For RC1

**Comment by Pilchie** [comment]
Approved for RC1 pending code review signoff if merged before 10am Pacific on 2020-09-01.

File: src/EFCore/Properties/CoreStrings.resx [hunk 1]
```diff
@@ -148,7 +148,7 @@
     <value>The service property '{property}' of type '{serviceType}' cannot be added to the entity type '{entityType}' because there is another property of the same type. Ignore one of the properties using the [NotMapped] attribute or 'EntityTypeBuilder.Ignore' in 'OnModelCreating'.</value>
   </data>
   <data name="AnnotationNotFound" xml:space="preserve">
-    <value>The annotation '{annotation}' was not found. Ensure that the annotation has been added.</value>
+    <value>The annotation '{annotation}' was not found. Ensure that the annotation has been added to the object {annotatable}</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:151]:
```suggestion
    <value>The annotation '{annotation}' was not found. Ensure that the annotation has been added to the object '{annotatable}'</value>
```

File: src/EFCore/Properties/CoreStrings.resx [hunk 2]
```diff
@@ -1329,7 +1326,10 @@
...
   </data>
   <data name="SelfReferencingNavigationWithInverseProperty" xml:space="preserve">
-    <value>A relationship cannot be established from property '{1_entityType}.{0_property}' to property '{3_referencedEntityType}.{2_referencedProperty}'. Check the values in the [InverseProperty] attribute to ensure relationship definitions are unique and reference from one navigation property to its corresponding inverse navigation property.</value>
+    <value>A relationship cannot be established from property '{1_entityType}.{0_property}' to property '{3_referencedEntityType}.{2_referencedProperty}'. Check the values in the [InverseProperty] attribute to ensure relationship definitions are unique and reference from one navigation to its corresponding inverse navigation.</value>
+  </data>
+  <data name="SensitiveDataDisabled" xml:space="preserve">
+    <value>To show additional information call EnableSensitiveDataLogging() when overriding DbContext.OnConfiguring.</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:1332]:
```suggestion
    <value>To show additional information call 'EnableSensitiveDataLogging()' when overriding 'DbContext.OnConfiguring'.</value>
```

File: src/EFCore/Properties/CoreStrings.resx [hunk 3]
```diff
@@ -1347,7 +1347,7 @@
     <value>configuration removed for '{key}'</value>
   </data>
   <data name="SetOperationWithDifferentIncludesInOperands" xml:space="preserve">
-    <value>When performing a set operation, both operands must have the same Include operations.</value>
+    <value>When performing a set operation, both operands must have the same 'Include' operations.</value>
```
Comment (smitpatel) [inline src/EFCore/Properties/CoreStrings.resx:1350]: We are somewhat inconsistent with Include in many places. Other places when referring to API uses `HasConversion()` with parenthesis.

File: src/EFCore/Properties/CoreStrings.resx (hunk 1 above)
Comment (AndriySvyryd) [reply src/EFCore/Properties/CoreStrings.resx:151]: `annotatable` could be a multi-line string

File: src/EFCore/Properties/CoreStrings.resx (hunk 3 above)
Comment (AndriySvyryd) [reply src/EFCore/Properties/CoreStrings.resx:1350]: Added to the list

**Comment by smitpatel** [comment]
Will PR checks be done in next 38 mins? 😆

**Comment by Pilchie** [comment]
Removing approval and auto-merge.  Please re-add `servicing-consider` and tag when ready to merge.

For RC1, this will require Tactics approval, or M2 approval for RC2.

**Comment by AndriySvyryd** [comment]
macOS timed out again 😢

**Comment by AndriySvyryd** [comment]
@Pilchie For RC2

**Comment by Pilchie** [comment]
Approved for RC2.

**Pull request merged**